tests/test_fee_burner.py

```diff
@@ -0,0 +1,377 @@
+    event_filter = fee_burner.events.ExchangeRateChanged.createFilter(fromBlock='latest')
+    # when: operator adds support for a new token
+    fee_burner.functions.addSupportFor(other_token.address, 1, 123).transact()
+
+    # then: ExchangeRateChanged event was emitted
+    emitted_events = event_filter.get_new_entries()
+
+    assert len(emitted_events) == 1
+    assert emitted_events[0]['args']['token'] == other_token.address
+    assert emitted_events[0]['args']['blockNo'] == w3.eth.blockNumber
+    assert emitted_events[0]['args']['nominator'] == 1
+    assert emitted_events[0]['args']['denominator'] == 123
+
+
+def test_events_emission_when_changing_rate(w3, fee_burner, other_token):
+    event_filter = fee_burner.events.ExchangeRateChanged.createFilter(fromBlock='latest')
+
+    # when: exchange rate is changed
+    fee_burner.functions.setExchangeRate(ZERO_ADDRESS, 11, 12).transact()
+
+    # then: ExchangeRateChanged event was emitted
+    emitted_events = event_filter.get_new_entries()
+
+    assert len(emitted_events) == 1
+    assert emitted_events[0]['args']['blockNo'] == w3.eth.blockNumber
+    assert emitted_events[0]['args']['nominator'] == 11
+    assert emitted_events[0]['args']['denominator'] == 12
+    assert emitted_events[0]['args']['token'] == ZERO_ADDRESS
+
+
+def test_set_new_rate_of_an_unsupported_token(fee_burner, operator, other_token):
+    # when: the operator tries to set a new exchange rate to a non existent token
+    # then: expect error
+    with pytest.raises(ValueError):
+        fee_burner.functions.setExchangeRate(other_token.address, 1, 1).transact()
+
+
+def test_setting_new_rate_when_maturity_period_has_not_passed(fee_burner, operator):
+    # given: pending exchange rate
+    fee_burner.functions.setExchangeRate(ZERO_ADDRESS, 11, 11).transact()
+
+    # when: the operator tires to change the rate one more
+    # then: expect error
+    with pytest.raises(ValueError):
+        fee_burner.functions.setExchangeRate(ZERO_ADDRESS, 22, 22).transact()
+
+
+def test_setting_new_rate_when_maturity_period_has_passed(w3, provider, fee_burner):
+    # given: pending exchange rate
+    fee_burner.functions.setExchangeRate(ZERO_ADDRESS, 11, 22).transact()
+    changed_rate = fee_burner.functions.getExchangeRate(ZERO_ADDRESS).call()
+
+    # when: maturity period has come
+    mine_blocks(provider, fee_burner.functions.NEW_RATE_MATURITY_MARGIN().call() + 1)
+    block_no = w3.eth.blockNumber
+
+    # then: operator can once more change the exchange rate
+    fee_burner.functions.setExchangeRate(ZERO_ADDRESS, 22, 33).transact()
+    assert fee_burner.functions.getExchangeRate(ZERO_ADDRESS).call() == [block_no + 1, 22, 33]
+    assert fee_burner.functions.getPreviousExchangeRate(ZERO_ADDRESS).call() == changed_rate
+
+
+def test_during_maturity_period_both_rates_should_be_valid(provider, non_operator, fee_burner, other_token, omg_token):
```

Comment:
you could add some tests where you test various combinations of rate_nominator, rate_denominator, omg_amount, other_amount - to make sure that the trades done always respect the exchange rate, i.e. when buing a token valued higher/lower/equal to omg